lib/web_ui/lib/src/engine/history/history.dart

```diff
@@ -25,29 +25,29 @@ abstract class BrowserHistory {
   late ui.VoidCallback _unsubscribe;
 
   /// The strategy to interact with html browser history.
-  LocationStrategy? get locationStrategy => _locationStrategy;
-  LocationStrategy? _locationStrategy;
+  JsUrlStrategy? get urlStrategy => _urlStrategy;
+  JsUrlStrategy? _urlStrategy;
```

Comment:
I feel like `BrowserHistory` should not be exposed to JS interop classes directly, but instead talk to the `UrlStrategy` type. More details in a comment for `convertToJsUrlStrategy`.